src/control/system/raft/database_members.go

```diff
@@ -47,28 +48,62 @@ func (mrm MemberRankMap) MarshalJSON() ([]byte, error) {
 	return json.Marshal(jm)
 }
 
-func (mam MemberAddrMap) addMember(addr *net.TCPAddr, m *system.Member) {
-	if _, exists := mam[addr.String()]; !exists {
-		mam[addr.String()] = []*system.Member{}
+// dedupeMembers removes members with duplicate UUIDs from MemberAddrMap.
+func (mam MemberAddrMap) dedupeMembers(addrStr string) {
+	seen := make(map[string]struct{})
+
+	newMembers := system.Members{}
+	for _, m := range mam[addrStr] {
+		us := m.UUID.String()
+
+		if _, exists := seen[us]; !exists {
```

Comment:
I am probably missing something but how do we know we are keeping the good member.

Reply:
The only plausible reason for having a duplicate is that identical member was added. We could verify all member fields before removing the duplicate and panic if not identical but I think this is overkill. @mjmac @kjacque thoughts?